Create Structure

yunuscanbartik opened the pull request.

yunuscanbartik committed:
Add NewsCatcherApi project with models and configurations

yunuscanbartik committed:
Configure Models and Add DB conneciton

yunuscanbartik committed:
Added Interfaces and edit structure

yunuscanbartik committed:
Added Services and Edit DbContext

yunuscanbartik committed:
Added Categories Service and Controllers

yunuscanbartik committed:
Add and Edit Services

yunuscanbartik committed:
Refactor, add APIs, and improve DI and Swagger config

Refactored `DatabaseContext` to use `IDatabaseContext` for better abstraction and modularity. Added new API controllers and endpoints for CRUD operations across multiple entities. Enhanced `CategoriesController` with additional endpoints. Introduced `SwashbuckleSchemaHelper` for custom Swagger schema ID generation and improved Swagger configuration. Standardized API responses for consistency. Removed unused code, improved dependency injection, and cleaned up formatting for better readability and maintainability.

yunuscanbartik committed:
Edit Services for read SqlData

yunuscanbartik committed:
Edit All Services structure. Add exception structure and tested

atakancelik commented on NewsCatcher.Services/Interfaces/INewsService.cs:
```diff
@@ -0,0 +1,13 @@
+        Task<NewsModel.BrowseModel.Return> GetNewsByIdAsync(NewsModel.BrowseModel.Request request);
+        Task<NewsModel.CreateModel.Return> AddNewsAsync(NewsModel.CreateModel.Request request);
+        Task<NewsModel.UpdateModel.Return> UpdateNewsAsync(NewsModel.UpdateModel.Request request);
+        Task<NewsModel.DeleteModel.Return> DeleteGetNewsAsync(NewsModel.DeleteModel.Request request);
```
bir onceki prdaki red sebeplerimi okuyup duzelt.

yunuscanbartik replied:
Düzeltildi.

atakancelik commented on NewsCatcher.Services/Services/CategoriesService.cs:
```diff
@@ -0,0 +1,274 @@
+            {
+                using (var reader = await sqlCommand.ExecuteReaderAsync()) //using'in amacı kullanıldıktan sonra otomatik kapatılması
+                {
+                    while (await reader.ReadAsync())
```
bu satirlardan once if hasrow ekleyebilirsin arastir.

yunuscanbartik replied:
Eklendi

atakancelik commented on NewsCatcher.Services/Services/NewsService.cs:
```diff
@@ -0,0 +1,283 @@
+                    ErrorCode = ex.HResult.ToString(),
+                    ErrorMessage = ex.Message,
+                    RequestId = Guid.NewGuid().ToString(),
+                    StatusCode = 404,
```
404 bulunamadi url hatasi demek bunu kullanmayalim. 200 donup status false ve errormessage vermemiz yeterli olur

yunuscanbartik committed:
Refactor models, services, and controllers for consistency

- Updated models to use nullable types for flexibility (`int?`, `string?`, etc.).
- Changed `Data` in `Return` classes to `List<ReturnData?>`.
- Fixed method names in service interfaces for clarity.
- Standardized `StatusCode` for error responses to `200` across services.
- Improved error messages for better user feedback.
- Fixed database read logic in `NotificationService` and `TagsService`.
- Corrected method call in `CategoriesController` to match service method.
- Performed general code cleanup for readability and maintainability.

yunuscanbartik committed:
Added hashRow

atakancelik reviewed (approved):
LGTM
